Library/Homebrew/cli_parser.rb

```diff
@@ -0,0 +1,48 @@
+require "optparse"
+require "ostruct"
+
+module Homebrew
+  module CLI
+    class Parser
+      def initialize(&block)
+        @parser = OptionParser.new
+        @parsed_args = OpenStruct.new
+        instance_eval(&block)
+      end
+
+      def switch(*names, description)
+        @parser.on(*names, description) do
+          names.each do |name|
+            @parsed_args["#{option_to_name(name)}?"] = true
+          end
+        end
+      end
+
+      def comma_array(name, description)
+        @parser.on(name, OptionParser::REQUIRED_ARGUMENT, Array, description) do |list|
+          @parsed_args[option_to_name(name)] = list
+        end
+      end
+
+      def flag(name, description, required: false)
```

Comment:
Maybe make description a named parameter i.e. `description:`?

Reply:
What differentiates a flag and a switch?

Reply:
`switch` is an option that does not take arguments. It is used to turn options on/off . It can be both long and short
Ex: --new-formula , -D

`flag`s take arguments. Long form flags take arguments with equal sign and short form flags don't require equal sign
Ex: --log-level=DEBUG